fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java

```diff
@@ -173,14 +238,213 @@ public ShowResultSet handleSubmitTaskStmt(SubmitTaskStmt submitTaskStmt) throws
+        String s = Text.readString(dis);
+        SerializeData data = GsonUtils.GSON.fromJson(s, SerializeData.class);
+        if (data != null) {
+            if (data.tasks != null) {
+                for (Task task : data.tasks) {
+                    replayCreateTask(task);
+                }
+            }
+            if (data.runStatus != null) {
+                for (TaskRunStatus runStatus : data.runStatus) {
+                    replayTaskRunCreateStatus(runStatus);
+                }
+            }
+        }
+    }
+
+    public List<TaskRunStatus> showTaskRunStatus(String dbName) {
+        List<TaskRunStatus> taskRunList = Lists.newArrayList();
+        if (dbName == null) {
+            for (Queue<TaskRun> pTaskRunQueue : taskRunManager.getPendingTaskRunMap().values()) {
+                taskRunList.addAll(pTaskRunQueue.stream().map(TaskRun::getStatus).collect(Collectors.toList()));
+            }
+            taskRunList.addAll(taskRunManager.getRunningTaskRunMap().values().stream().map(TaskRun::getStatus)
+                    .collect(Collectors.toList()));
+            taskRunList.addAll(taskRunManager.getTaskRunHistory().getAllHistory());
+        } else {
+            for (Queue<TaskRun> pTaskRunQueue : taskRunManager.getPendingTaskRunMap().values()) {
+                taskRunList.addAll(pTaskRunQueue.stream().map(TaskRun::getStatus)
+                        .filter(u -> u.getDbName().equals(dbName)).collect(Collectors.toList()));
+            }
+            taskRunList.addAll(taskRunManager.getRunningTaskRunMap().values().stream().map(TaskRun::getStatus)
+                    .filter(u -> u.getDbName().equals(dbName)).collect(Collectors.toList()));
+            taskRunList.addAll(taskRunManager.getTaskRunHistory().getAllHistory().stream()
+                    .filter(u -> u.getDbName().equals(dbName)).collect(Collectors.toList()));
+
+        }
+        return taskRunList;
+    }
+
+    public void replayTaskRunCreateStatus(TaskRunStatus status) {
+
+        switch (status.getState()) {
+            case PENDING:
+                String taskName = status.getTaskName();
+                Task task = nameToTaskMap.get(taskName);
+                if (task == null) {
+                    return;
+                }
+                TaskRun taskRun = TaskRunBuilder.newBuilder(task).build();
+                taskRun.initStatus(status.getQueryId(), status.getCreateTime());
+                Queue<TaskRun> taskRuns = taskRunManager.getPendingTaskRunMap().computeIfAbsent(taskRun.getTaskId(),
+                        u -> Queues.newConcurrentLinkedQueue());
+                taskRuns.offer(taskRun);
+                break;
+            // this will happen in build image
+            case RUNNING:
+                status.setState(Constants.TaskRunState.FAILED);
+                taskRunManager.getTaskRunHistory().addHistory(status);
+                break;
+            case FAILED:
+            case SUCCESS:
```

Comment:
Add the clear logic here. To avoid there are too many finished tasks loaded to memory.